Libraries/LibCore/ProcessStatisticsReader.cpp

```diff
@@ -108,6 +109,43 @@ HashMap<pid_t, Core::ProcessStatistics> ProcessStatisticsReader::get_all()
     return map;
 }
 
+Optional<String> ProcessStatisticsReader::get_current_working_directory(pid_t pid)
+{
+    StringBuilder builder;
+    builder.append("/proc/");
+    builder.append(String::number(pid));
+    builder.append("/cwd");
+
+    char buffer[PATH_MAX];
+    int length = readlink(builder.to_string().characters(), buffer, sizeof(buffer));
+    if (length < 0) {
+        perror("readlink");
+    } else {
+        ASSERT(length > 0);
+        return String(buffer, length);
+    }
```

Comment:
Please add a convenience API like `static String Core::File::readlink(const String& path)` for this so we don't have to see this awkward C business more than necessary. :)